### PR TITLE
Allowed configuration of pkgs_dirs.

### PR DESCRIPTION
**Pull request by pelson**
Allows either site (sys_rc) or user (rc) configuration of the pkgs directories using the ``pkgs_dirs`` section in the yaml config.

### CODE REVIEW

File: conda/config.py [hunk 1]
```diff
@@ -149,7 +149,17 @@ def pkgs_dir_from_envs_dir(envs_dir):
     else:
         return join(envs_dir, '.pkgs')
 
-pkgs_dirs = [pkgs_dir_from_envs_dir(envs_dir) for envs_dir in envs_dirs]
+
+def _get_pkgs_dirs(envs_dirs):
+    if 'pkgs_dirs' in rc:
+        pkgs = rc['pkgs_dirs']
+    elif 'pkgs_dirs' in sys_rc:
+        pkgs = sys_rc['pkgs_dirs']
```
Comment (pelson) [inline conda/config.py]: Should probably normalize these.

**Comment by tswicegood** [comment]
@pelson if you merge `master` into this branch and re-push it should fix the failing test case.  We had an issue that I just resolved around TravisCI and Python 3.3 test errors.

**Comment by tswicegood** [comment]
I'm :+1: on this.  I would really like to see some unit tests added around it before merging.  Not to check the logic but to provide a safety net and ensure API stability going forward.  Do you have the bandwidth to take a stab at that?

Testing will probably require some mocking.  Once I merge #1045 (coming shortly) we'll have some testing utilities in place to handle mocking and skipping tests if mock isn't available.

**Comment by pelson** [comment]
> Do you have the bandwidth to take a stab at that?

Yes, I'll take a look. I was curious how testing would take place with conda anyway.

**Comment by tswicegood** [comment]
Awesome!  This one's going to be a little difficult because it's testing functionality that's set at module load.  Off the top of my head, I'd test that the variables equals the output of the function, then test the function directly with `mock.patch.object` swapping out the various other bits it relies on.

**Comment by ilanschnell** [comment]
I don't see any reason to make the package cache location configurable.

**Pull request closed without merging**